jax/_src/numpy/lax_numpy.py

```diff
@@ -1368,6 +1368,82 @@ def gradient(f, *varargs, axis: Optional[Union[int, Tuple[int, ...]]] = None,
 def isrealobj(x):
   return not iscomplexobj(x)
 
+@partial(jit, static_argnums=1)
+class _RankWarning(UserWarning):
+    """
+    Issued by `polyfit` when the Vandermonde matrix is rank deficient.
+    For more information, a way to suppress the warning, and an example of
+    `RankWarning` being issued, see `polyfit`.
+    """
+    pass
+
+@_wraps(np.polyfit)
+def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
+  """
+  Note: Unlike numpy, the function will not warn on rank reduction, which indicates an ill conditioned matrix
+  """
```

Comment:
This docstring will be overwritten by `_wraps`. You can put this info in the `lax_description` argument of `_wraps`; there are a few examples elsewhere in this file.